lib/rescue_unique_constraint.rb

```diff
@@ -1,34 +1,93 @@
-require "rescue_unique_constraint/version"
+require 'rescue_unique_constraint/version'
 require 'active_record'
 
+# Module which will rescue ActiveRecord::RecordNotUnique exceptions
+# and add errors for indexes that are registered with
+# rescue_unique_constraint(index:, field:)
 module RescueUniqueConstraint
+  # handles storing and matching [index, field] pairs to exceptions
+  class RescueHandler
```

Comment:
seems like we should have a handler per adapter instead of the different sqllite and postres methods.